Migrate history files (mg5history, me5history, mw5history, decayhistory) to ensure XDG Base Directory compliance

Bruno-M-S-Oliveira opened the pull request:
This update involves moving the history files (mg5history, me5history, mw5history, decayhistory) used by mg5amcnlo to a new directory called "$XDG_STATE_HOME/mg5/" to conform with the XDG Base Directory Specification. The change will only take place if the user's $HOME/.mg5/ directory doesn't already exist. This ensures backward compatibility and prevents data loss or confusion for existing users.

The update is meant to improve the user experience by ensuring that critical state data is stored in a location that is accessible and portable and follows a widely recognized standard. 

The configuration file $HOME/.mg5/mg5_configuration.txt could also be migrated to $XDG_CONFIG_HOME/mg5/mg5_configuration.txt, but since it involves more complicated files, such as madgraph/interface/{extended_cmd.py, extended_cmd.py, madgraph_interface.py, madgraph_interface.py, common_run_interface.py}, I'd prefer to have approval before proceeding.

Furthermore, to prevent redundant code, it would be best to store the state_dir (and potentially config_dir) as a global variable, but I'm not sufficiently proficient in Python to do so without some assistance.

Bruno-M-S-Oliveira committed:
Change history files (mg5history, me5history, mw5history, decayhistory) from $HOME/.mg5/ to $XDG_STATE_HOME/mg5/, unless $HOME/.mg5/ already exists.

According to the XDG Base Directory specification (https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html):
> The $XDG_STATE_HOME contains state data that should persist between (application) restarts, but that is not important or portable enough to the user that it should be stored in $XDG_DATA_HOME. It may contain:
>
>    actions history (logs, history, recently used files, …)
>
>    current state of the application that can be reused on a restart (view, layout, open files, undo history, …)

oliviermattelaer commented:
Should not we also have some hack around those to met the specification:
```
[madgraph]$ grep mg5_configuration . -rin | grep "\.mg5"
./interface/madgraph_interface.py:320:        logger.info('   If you put "global" for FILENAME it will use ~/.mg5/mg5_configuration.txt')
./interface/madgraph_interface.py:1470:                        args.insert(1,pjoin(os.environ['HOME'],'.mg5','mg5_configuration.txt'))
./interface/madgraph_interface.py:3072:        # Load the configuration file,i.e.mg5_configuration.txt
./interface/madgraph_interface.py:6014:            config_file = '~/.mg5/mg5_configuration.txt'
./interface/madgraph_interface.py:7176:            files.cp(pjoin(MG5DIR,'input','.mg5_configuration_default.txt'), config_path)
./interface/madgraph_interface.py:7660:                basefile = pjoin(MG5DIR, 'input', '.mg5_configuration_default.txt')
./interface/extended_cmd.py:1956:        os.path.exists(pjoin(os.environ['HOME'], '.mg5', 'mg5_configuration.txt')):
./interface/extended_cmd.py:1957:            base = pjoin(os.environ['HOME'], '.mg5', 'mg5_configuration.txt')
```

Bruno-M-S-Oliveira committed:
Change config file (mg5_configuration.txt) from $HOME/.mg5/ to $XDG_CONFIG_HOME/mg5/, unless $HOME/.mg5/ already exists.

According to the XDG Base Directory specification (https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html):
>  $XDG_CONFIG_HOME defines the base directory relative to which user-specific configuration files should be stored. If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used.

Bruno-M-S-Oliveira commented:
I was just waiting for your approval. The latest commit migrates those as well.

oliviermattelaer commented:
So this is now merged within 3.5.0 via this commit: 
https://github.com/mg5amcnlo/mg5amcnlo/commit/118ef6ea061ad82a4720a1c353d402646f6c0e44
Which is basically a squash of your two commit and removal all formating change to file (to avoid spurious conflict in the future).

Thanks,

Olivier
